keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1196 @@
+If there is no capacity defined, the device is considered as having an **infinite** sharable capacity.
+
+Users can define specific per-device resource requests through
+the newly added `ResourceRequest` field in the `DeviceRequest` of the `ResourceClaim`.
+In a resource request, users may define `Requests` and `Limits` or either of them
+to specify the claim's requirements. Similarly to the node resources, a request must be less than its limit.
+If either value is omitted, the other value will be implicitly applied. 
+`Requests` is used for verifying consumability of the shared device
+while `Limits` is optionally used for supporting burstable consumption controlled by the device driver.
+
+To request all available resources in the device capacity, users must set a flag `all` to "true".
+Regardless of whether the shared device has infinite sharable capacity or not,
+this flag will block the allocation of the shared device from the other claim.
+
+If the `resources` field has been defined, only shared device can be allocated.
+On the other hand, if the `resources` field is nil, only non-sharable devices are expected
+except the allocation mode `AllowShared` has been set.
+For allocationMode=AllowShared, both non-sharable devices and shared devices can be allocated.
+
+A shared device can only be allocated once its consumability has been verified
+and its attributes match the request's selectors and constraints.
+The newly added `resources` field in the `DeviceRequestAllocationResult` will be set when the allocation is successful.
+A device can be shared (allocated) to different pods. 
+However, claims from the same pod cannot allocate shares from the same shared device.
+
+### API enhancement
+To enable this enhancement, the following API updates are proposed.
+#### ResourceSliceSpec's BasicDevice
+
+```go
+
+// BasicDevice defines one device instance.
+type BasicDevice struct {
+...
+   // Shared marks whether the device is shared.
+   // The device with shared="true" can be allocated to more than one claim,
+   // and all value in capacity is considered as consumable.
+   // If there is no capacity defined,
+   // the device is considered as having an infinity sharable capacity.
+   // +optional
+   // +default=false
+   // +featureGate=ConsumableCapacity
+   Shared bool `json:"shared" protobuf:"bytes,3,opt,name=shared"`
+}
+```
+
+#### ResourceClaimSpec's DeviceRequest
+
+```go
+// ResourceRequest is a per-device resource request specification.
+type ResourceRequest struct {
+   // All marks requesting all resources from the shared device.
+   // Regardless of the number of capacity defined in the shared device, 
+   // this flag will block the allocation of the shared device from the other claims.
+   // +optional
+   // +default=false
+   All bool `json:"all" protobuf:"bytes,1,name=all"`
+
+
+   // Requirements define specific values of resource requirements of the device request.
+   // If all="true", this field is ignored.
```

Comment:
Rather than saying the field will be ignored, we want to validate that only one of the two fields are set.

Reply:
API design changed and all field is removed.